pkg/broker/provision_subscriber.go

```diff
@@ -0,0 +1,41 @@
+package broker
+
+import (
+	"encoding/json"
+
+	"github.com/fusor/ansible-service-broker/pkg/apb"
+	"github.com/fusor/ansible-service-broker/pkg/dao"
+)
+
+type ProvisionWorkSubscriber struct {
+	dao       *dao.Dao
+	msgBuffer <-chan WorkMsg
+}
+
+func NewProvisionWorkSubscriber(dao *dao.Dao) *ProvisionWorkSubscriber {
+	return &ProvisionWorkSubscriber{dao: dao}
+}
+
+func (p *ProvisionWorkSubscriber) Subscribe(msgBuffer <-chan WorkMsg) {
+	p.msgBuffer = msgBuffer
+
+	var pmsg *ProvisionMsg
+	var extCreds *apb.ExtractedCredentials
+	go func() {
+		for {
```

Comment:
Yeah, I think I got bit by the cargo cult. This was a long lived subscriber that wrote to websocket. I wrote a quick prototype to use this and sure enough it'll hang forever. :( Working on a fix. Good catch.